lib/core/shards/usecase.ts

```diff
@@ -0,0 +1,51 @@
+import { MirrorsRepository } from '../mirrors/Repository';
+import { ShardsRepository } from './Repository';
+import NetworkMessageQueue from "../../server/queues/networkQueue";
+import { DELETING_FILE_MESSAGE } from "../../server/queues/messageTypes";
+import { BucketEntryVersionNotFoundError } from '../bucketEntries/usecase';
+
+export class ShardsUsecase {
+  constructor(
+    private shardsRepository: ShardsRepository,
+    private mirrorsRepository: MirrorsRepository,
+    private networkQueue: NetworkMessageQueue
+  ) {}
+
+  enqueueDeleteShardMessages = async(hashes: string[], version: number) => {
```

Comment:
Maybe hiding if is being enqueued, cached, listed, or inserted could be a better idea when naming is used for the use-case. 

`deleteShard(s)` could be more business-logic oriented than a concrete naming suggesting a specification of the implementation done